pkg/operator/controller.go

```diff
@@ -135,6 +139,81 @@ func (c *Controller) createOrUpdateResources(cr *imageregistryv1.Config) error {
 }
 
 func (c *Controller) sync() error {
+	prunerResourceFound := true
+	pcr, err := c.listers.ImagePrunerConfigs.Get(defaults.ImageRegistryImagePrunerResourceName)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			prunerResourceFound = false
+		} else {
+			return fmt.Errorf("failed to get %q image registry pruner resource: %s", defaults.ImageRegistryImagePrunerResourceName, err)
+		}
+	}
+	if prunerResourceFound {
+		pcr = pcr.DeepCopy() // we don't want to change the cached version
+		prevPCR := pcr.DeepCopy()
+		prunerCronJob, err := c.listers.CronJobs.Get("image-pruner")
+		if errors.IsNotFound(err) {
+			prunerCronJob = nil
+		} else if err != nil {
+			return fmt.Errorf("failed to get image-pruner pruner job: %s", err)
+		} else {
+			prunerCronJob = prunerCronJob.DeepCopy()
+		}
+
+		prunerJobs, err := c.listers.Jobs.List(labels.Everything())
+		if err != nil {
+			return fmt.Errorf("failed to get pruner jobs: %s", err)
+		}
+		lastPrunerJobConditions := []batchv1.JobCondition{}
+		if len(prunerJobs) > 0 {
+			lastPrunerJobConditions = prunerJobs[len(prunerJobs)-1].Status.Conditions
```

Comment:
Are we guaranteed to get the list of `Jobs` in their order of creation, or some other value?

Reply:
I'm not sure, i don't see anything in the docs about what order they are in.  I thought we ran into this somewhere else in the code also, let me see if i can find it.

Reply:
Ah, I think it was pruning builds.  Let me go look there.

Reply:
Added a sort by creation timestamp in current push.